financialaid/views_test.py

```diff
@@ -269,3 +246,277 @@ def test_review_financial_aid_view_with_invalid_filter_and_sorting(self):
+        )
+        assert resp.status_code == status.HTTP_403_FORBIDDEN
+        # Not allowed for logged-out user
+        self.client.logout()
+        resp = self.client.put(
+            self.action_url,
+            data=json.dumps(self.financial_review_data),
+            content_type="application/json"
+        )
+        assert resp.status_code == status.HTTP_403_FORBIDDEN
+
+    def test_invalid_action(self, *args):  # pylint: disable=unused-argument
+        """
+        Tests FinancialAidActionView when invalid action is posted
+        """
+        # Invalid action
+        self.client.force_login(self.staff_user_profile.user)
+        self.financial_review_data["action"] = FinancialAidStatus.PENDING_DOCS
+        resp = self.client.put(
+            self.action_url,
+            data=json.dumps(self.financial_review_data),
+            content_type="application/json"
+        )
+        assert resp.status_code == status.HTTP_400_BAD_REQUEST
+
+    def test_invalid_tier_program(self, *args):  # pylint: disable=unused-argument
+        """
+        Tests FinancialAidActionView when invalid tier_program is posted
+        """
+        # Not current
+        self.client.force_login(self.staff_user_profile.user)
+        self.financial_review_data["action"] = FinancialAidStatus.APPROVED
+        self.financial_review_data["tier_program_id"] = self.tier_programs["150k_not_current"].id
+        resp = self.client.put(
+            self.action_url,
+            data=json.dumps(self.financial_review_data),
+            content_type="application/json"
+        )
+        assert resp.status_code == status.HTTP_400_BAD_REQUEST
+        # Not part of the same program
+        new_tier_program = TierProgramFactory.create()  # Will be part of a different program
+        self.financial_review_data["tier_program_id"] = new_tier_program.id
+        resp = self.client.put(
+            self.action_url,
+            data=json.dumps(self.financial_review_data),
+            content_type="application/json"
+        )
+        assert resp.status_code == status.HTTP_400_BAD_REQUEST
+
+    def test_reject_invalid_status(self, *args):  # pylint: disable=unused-argument
+        """
+        Tests FinancialAidActionView when trying to reject a FinancialAid that isn't pending manual approval
+        """
+        # FinancialAid object that cannot be rejected
+        self.client.force_login(self.staff_user_profile.user)
+        self.financialaid.status = FinancialAidStatus.PENDING_DOCS
+        self.financialaid.save()
+        self.financial_review_data["action"] = FinancialAidStatus.REJECTED
+        self.financial_review_data["tier_program_id"] = self.tier_programs["15k"].id
+        resp = self.client.put(
+            self.action_url,
+            data=json.dumps(self.financial_review_data),
+            content_type="application/json"
+        )
+        assert resp.status_code == status.HTTP_400_BAD_REQUEST
+        # Reset FinancialAid object
```

Comment:
Moved.  (Also, would you prefer that we respond to all of your comments so you know we're acknowledging it, or would you prefer that we spare your inbox/notification queue?)